BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -319,20 +318,57 @@ actor DefaultAuthenticatorSyncService: NSObject, AuthenticatorSyncService {
     ///   - userId: The userId of the account to which the Ciphers belong.
     ///
     private func writeCiphers(ciphers: [Cipher], userId: String) async throws {
-        let account = try await stateService.getAccount(userId: userId)
-        let useKey = await vaultTimeoutService.isLocked(userId: userId)
-
         do {
-            if useKey {
-                try await authRepository.unlockVaultWithAuthenticatorVaultKey(userId: userId)
-            }
+            try await unlockVaultWithAuthenticatorVaultKey(userId: userId)
+
+            let account = try await stateService.getAccount(userId: userId)
             let items = try await decryptTOTPs(ciphers, account: account)
             try await authBridgeItemService.replaceAllItems(with: items, forUserId: userId)
+
+            try await clientService.removeClient()
         } catch {
             errorReporter.log(error: error)
+            try? await clientService.removeClient()
         }
-        if useKey {
-            await vaultTimeoutService.lockVault(userId: userId)
-        }
+    }
+
+    // MARK: Vault Unlocking
+
+    /// Initializes the SDK's crypto for any organizations the users is a member of.
+    ///
+    /// - Parameter userId: The userId the account being used to initialize the SDK.
+    ///
+    private func initializeOrganizationCrypto(userId: String) async throws {
+        let organizationKeysById = try await organizationService.fetchAllOrganizations(userId: userId)
+            .reduce(into: [String: String]()) { result, organization in
+                guard let key = organization.key else { return }
+                result[organization.id] = key
+            }
+        try await clientService.crypto().initializeOrgCrypto(
+            req: InitOrgCryptoRequest(organizationKeys: organizationKeysById)
+        )
+    }
```

Comment:
🤔 Unless I'm missing something this is the same as the one in `OrganizationService` but with a specific `userId`:
https://github.com/bitwarden/ios/blob/f4f2fe624c86c7948a8d341f0bb76136f168d6a7/BitwardenShared/Core/Vault/Services/OrganizationService.swift#L85-L107

What do you think about reusing the same as in `OrganizationService` but passing a `userId: String? = nil`? So if it's `nil` then it grabs the active user and if not it uses the parameter one which would be for this case.